src/operators.rs

```diff
@@ -15,12 +15,12 @@ pub trait BinaryOperator<V: Value>: Debug + Clone {
     fn apply_2(&self, a: &V, b: &V) -> V;
 }
 
-pub trait Initializer<V: Value>: Debug + Clone {
-    fn initialize(&self, values: Vec<V>) -> V;
+pub trait Initializer<TS: crate::TypeSystem>: Debug + Clone {
+    fn initialize(&self, values: Vec<TS::Value>, ctx: &mut TS::GlobalContext) -> TS::Value;
```

Comment:
I would just pass `&mut ExecutionEngine`, since that includes the context anyways and that's what's passed to native functions